# Crossmatch habitable zone targets with the CKS sample and my own rotation period catalogue.

In [1]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import pandas as pd

Load Stephen Kane's list of targets, the CKS catalogue, the McQuillan catalogue and my own rotation period catalogue.

In [14]:
# Kane catalogue:
kane = pd.read_csv("kane_targets.csv")

# CKS catalogue:
cks = pd.read_csv("cks_physical_merged.csv")

# McQuillan catalogue:
mc = pd.read_csv("Table_1_Periodic.txt")

# Tim and my catalogue:
tdmra = pd.read_csv("koi_periods_0712.csv")

What's the overlap between Stephen Kane's catalogue and the CKS sample?

In [44]:
# Format the CKS catalogue and Kane catalogue column headers.
koi_ids = []
for i in cks.id_koicand:
    koi_ids.append("Kepler-{}".format(int(float(i[1:]))))
cks["koi_id"] = koi_ids

KOIs = np.zeros(np.shape(kane)[0], dtype=int)
for j, i in enumerate(kane.koi_id):
    if i[:7] == "Kepler-":
        KOIs[j] = int((float(i[7:])))

kane["KOI"] = KOIs
print(kane.keys(), "\n")
print(cks.keys())
kane_cks = pd.merge(kane, cks, on="koi_id", how="inner")
print(np.shape(kane), np.shape(cks), np.shape(kane_cks))

Index(['koi_id', 'planet_designation', 'p_orb_days', 'a_AU', 'r_pl_earth',
       'teff', 'r_star_rsol', 'insolation_flux_F_E', 'KOI'],
      dtype='object') 

Index(['id_starname', 'id_kic', 'id_koicand', 'id_kepler_name',
       'koi_disposition', 'koi_period', 'koi_period_err1', 'koi_period_err2',
       'koi_time0', 'koi_time0_err1',
       ...
       'iso_sma', 'iso_sma_err1', 'iso_sma_err2', 'iso_insol',
       'iso_insol_err1', 'iso_insol_err2', 'iso_teq', 'iso_teq_err1',
       'iso_teq_err2', 'koi_id'],
      dtype='object', length=121)
(1162, 9) (2025, 121) (617, 129)


617 stars have CKS spectra. ,How many of these stars have rotation periods measured by Amy McQuillan?

In [65]:
kane_cks_mc = pd.merge(kane_cks, mc, left_on="id_kic", right_on="kepid", how="inner")
np.shape(kane_cks_mc)

(0, 140)

None... How about in Tim and my catalogue?

In [64]:
kane_cks_tdmra = pd.merge(kane_cks, tdmra, on="KOI", how="inner")
np.shape(kane_cks_tdmra)

(552, 142)

So there are 552 stars with spectroscopic properties AND rotation periods. The next thing to do is to combine all this information using isochrones and a gyrochronology model. See the ipython notebook called age_inference.ipynb for that.